### ISSUE TITLE
file_rec/async:! extremely slow

### ISSUE THREAD

**Issue by carlesandres** [Shougo/unite.vim#723]
- Problems summary
  When I invoke `Unite -no-split -buffer-name=files   -start-insert file_rec/async:!` and I start typing a file name, it takes seconds to start to showing results. Moreover, despite having selected `Ag` as a my grep file, all the files in my `.gitignore` file still get in the list. After 10 more seconds in the list, Vim gets completely blocked for more 30seconds.
  
  The first I invoke the function it works as expected, but the from second one it is reaaaaally slow.
- Expected
  You know, smooth and fast behaviour.
- Environment Information
  - OS: 10.9.4
  - Vim version: 7.4
- Minimal vimrc less than 50 lines

```
" Your vimrc
" Set up some custom ignores
call unite#custom_source('file_rec,file_rec/async,file_mru,file,buffer,grep',
      \ 'ignore_pattern', join([
      \ '\.git/',
      \ ], '\|'))

if executable('ag')
  let g:unite_source_rec_async_command= 'ag --follow --nocolor --nogroup --hidden -g ""'
  let g:unite_source_grep_command = 'ag'
  let g:unite_source_grep_default_opts = '--column --nogroup --nogroup'
  let g:unite_source_grep_recursive_opt = ''
endif

" let g:unite_source_rec_max_cache_files = 99999

" See https://github.com/Shougo/unite.vim/issues/236#issuecomment-51983184
let g:unite_source_rec_unit = 250

let g:unite_source_history_yank_enable = 1
" call unite#filters#sorter_default#use(['sorter_rank'])
call unite#filters#matcher_default#use(['matcher_fuzzy'])
nnoremap <leader>h :<C-u>Unite -no-split -buffer-name=files   -start-insert file_rec/async:!<cr>
nnoremap <leader>F :<C-u>Unite -no-split -buffer-name=files   -start-insert file<cr>
nnoremap <leader>r :<C-u>Unite -no-split -buffer-name=mru     -start-insert file_mru<cr>
nnoremap <leader>y :<C-u>Unite -no-split -buffer-name=yank    history/yank<cr>
nnoremap <leader>e :<C-u>Unite -no-split -buffer-name=buffer  buffer<cr>
nnoremap <leader>b :<C-u>Unite -no-split -buffer-name=buffer  -start-insert buffer<cr>
nnoremap <leader>G :<C-u>Unite -no-split grep:.<cr>
```
- Reproducable ways from Vim starting

I don't really quite get what this means.

 0.startup vim.
 1.
 2.
 3.
- Screen shot (if possible)

**Comment by Shougo** [comment]
Fixed in Vim 7.4.615

https://groups.google.com/forum/#!topic/vim_dev/pXjj7rww5RY

**Comment by carlesandres** [comment]
Great. Thanks for the info.
